target/cheri-common/cheri-compressed-cap/cheri_compressed_cap_common.h

```diff
@@ -102,16 +105,26 @@ struct _cc_N(cap) {
     inline uint32_t type() const { return cr_otype; }
     inline bool is_sealed() const { return cr_otype != _CC_N(OTYPE_UNSEALED); }
     inline bool operator==(const _cc_N(cap) & other) const {
-        return _cr_cursor == other._cr_cursor && cr_base == other.cr_base && _cr_top == other._cr_top &&
-               cr_perms == other.cr_perms && cr_uperms == other.cr_uperms && cr_otype == other.cr_otype &&
-               cr_ebt == other.cr_ebt && cr_flags == other.cr_flags && cr_reserved == other.cr_reserved &&
-               cr_tag == other.cr_tag;
+      return _cc_N(exactly_equal)(this, other);
```

Comment:
I'd move the implementation out of line to avoid the forward declaration. I think there's another ifdef c++ further down where this could go. But I don't really mind this is fine too.